features/networking/multus.feature

```diff
@@ -1394,4 +1394,55 @@ Feature: Multus-CNI related scenarios
+      
+  # @author weliang@redhat.com
+  # @case_id OCP-31999
+  @admin
+  Scenario: Whereabouts with exclude IP address	
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    Given I obtain test data file "networking/multus-cni/NetworkAttachmentDefinitions/whereabouts-excludeIP.yaml"
+    When I run oc create as admin over "whereabouts-excludeIP.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %> |
+    Then the step should succeed
+    
+    # Create a pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | macvlan-bridge-whereabouts-pod1 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip           |
+      | ["spec"]["containers"][0]["name"]                          | whereabouts-excludeip           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod1" becomes ready
+    # Check the created pod has correct ip 
+    When I execute on the "macvlan-bridge-whereabouts-pod1" pod:
+      | bash | -c | /usr/sbin/ip -4 -brief a |
+    Then the output should contain:
+      | 192.168.10.4 |
+
+    # Create second pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | macvlan-bridge-whereabouts-pod2 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip           |
+      | ["spec"]["containers"][0]["name"]                          | whereabouts-excludeip           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod2" becomes ready
+    # Check the created pod has correct ip 
+    When I execute on the "macvlan-bridge-whereabouts-pod2" pod:
+      | bash | -c | /usr/sbin/ip -4 -brief a |
+    Then the output should contain:
+      | 192.168.10.5 |
+      
+    # Create third pod absorbing above net-attach-def
+    Given I obtain test data file "networking/multus-cni/Pods/generic_multus_pod.yaml"
+    When I run oc create over "generic_multus_pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                       | macvlan-bridge-whereabouts-pod3 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | whereabouts-excludeip           |
+      | ["spec"]["containers"][0]["name"]                          | whereabouts-excludeip           |
+    Then the step should succeed
+    And the pod named "macvlan-bridge-whereabouts-pod3" status becomes :pending within 60 seconds
```

Comment:
is there any logs can be shown or `oc describe pod` can list some information about `no available ip`

Reply:
@zhaozhanqi Add the step to check "Could not allocate IP in range" message shown in pod describe log. Thanks!

New running log:
https://privatebin-it-iso.int.open.paas.redhat.com/?fbc8d991b7dea42d#M051Eue3gU1J9sF1UBeTqqgo/AZ6RrIeLQb4+yDrfyk=

Reply:
thanks, LGTM